core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
> Since its only use right now is comparing two environments within Node, I don't agree.

That's clear and noted. You said that to keep the check here for the reason if we add more fields we won't forget them in the assert you would like more checks. Can you make a suggestion as to what to add?

To be honest, I added these because I'm conflicted between two things:

1) I don't want the Environment to expose a different settings than the one the node has.
2) I don't want the setting in the Environment to reflect something else then was actually used in the security manager, for example. 

I chose to reinitialize the Environment with the new settings and make sure nothing of the things that matter changed (it currently can't so I added an assertion).

I will try to another route and remote the Environment.getSettings() method, which means all of this goes away. I will do it in another PR